br/pkg/lightning/backend/local/local.go

```diff
@@ -735,12 +740,21 @@ func (local *Backend) lockAllEnginesUnless(newState, ignoreStateMask importMutex
 
 // Close the local backend.
 func (local *Backend) Close() {
-	allEngines := local.lockAllEnginesUnless(importMutexStateClose, 0)
+	allEngines := make([]common.Engine, 0, 8)
+	for _, e := range local.externalEngine {
+		allEngines = append(allEngines, e)
```

Comment:
Why not close it directly?

Reply:
https://github.com/pingcap/tidb/pull/46636/commits/4a8925702248bbb4c5af0c99574b117a13c0b266